PortMaster/mapper.py

```diff
@@ -72,76 +72,96 @@ def premap_input(input_entry):
         rightanalogdown = map_input("rightanalogdown", input_type, input_id, invert_value)
         return f"{rightanalogup}{rightanalogdown}"
     else:
+         # button or hat
          return f"{map_input(input_name, input_type, input_id, input_value)}"
 
 # Map the actual button/hat/axis
 def map_input(input_name, input_type, input_id, input_value):
 
-
+    # Ignore unknown ES input name
     if not input_name in TR_MAP.keys():
       print(f"Invalid mapping {input_name}.")
       return ""
 
+    # Map ES input name to SDL controller input name
     tr_name = TR_MAP[input_name]
 
     print(f"{input_name} -> {tr_name}")
 
+    # Axis can be either a stick or an analog trigger
     if input_type == "axis":
         if int(input_value) < 0:
+            # Stick (negative)
             return f"{tr_name}:-a{input_id},"
         else:
             # Most (save for a few misbehaved children...) triggers are [0, 1] instead of [-1, 1]
             # Shitty workaround for an emulationstation issue
             if "trigger" in input_name:
+                # Analog trigger (full range)
                 return f"{tr_name}:a{input_id},"
             else:
+                # Stick (positive)
                 return f"{tr_name}:+a{input_id},"
 
     elif input_type == "button":
+        # Button
         return f"{tr_name}:b{input_id},"
     
     elif input_type == "hat":
+        # Hat
         return f"{tr_name}:h{input_id}.{input_value},"
     
     else:
+        # Ignore unknow type
         print(f"Invalid entry {input_type}")
         return ""
 
 def main():
 
+    default_es_input_path = Path.home() / ".config" / "emulationstation" / "es_input.cfg"
     parser = argparse.ArgumentParser(description='ES input to gamecontrollerdb mapper')
-    parser.add_argument('filepath', help='gamecontrollerdb file path (eg: /tmp/gamecontrollerdb.txt)')
+    parser.add_argument('-i','--input', default=f"{default_es_input_path}", help='es_input file path (eg: ${HOME}/.config/emulationstation/es_input.cfg)')
+    parser.add_argument('gamecontrollerdb_path', help='gamecontrollerdb file path (eg: /tmp/gamecontrollerdb.txt)')
 
     args = parser.parse_args()
 
+    # List of already configured device GUID
     deviceGUID_list = []
 
-    with open(args.filepath,"r") as gamecontrollerdb:
-        for line in gamecontrollerdb.readlines():
-            line = line.strip()
-            if line.startswith("#") or len(line) == 0:
-                continue
-            deviceGUID = line.split(",")[0]
-            deviceGUID_list.append(deviceGUID)
-
-    es_input_path = Path.home() / "configs" / "emulationstation" / ES_INPUT_CFG
-
-    if not es_input_path.is_file():
-        es_input_path = Path.home() / ".config" / "emulationstation" / ES_INPUT_CFG
```

Comment:
The path for the file should be constructed before calling the mapper script. Anyway now we only call it in `PortMaster/control.txt` for one path :)